controllers/propagator/encryption_test.go

```diff
@@ -103,3 +106,51 @@ func TestGetEncryptionKeyCached(t *testing.T) {
 	Expect(err).To(BeNil())
 	Expect(string(key[0])).To(Equal("A"))
 }
+
+func TestGetInitializationVector(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	// Test when the initialization vector is generated
+	tests := []struct {
+		description string
+		annotations map[string]string
+	}{
+		{
+			"No IV",
+			map[string]string{},
+		},
+		{
+			"Valid IV",
+			map[string]string{
+				ivAnnotation: "7cznVUq5SXEE4RMZNkGOrQ==",
+			},
+		},
+		{
+			"Invalid IV",
+			map[string]string{
+				ivAnnotation: "this-is-invalid",
+			},
+		},
+	}
+
+	r := PolicyReconciler{}
+
+	for _, test := range tests {
+		t.Run(
```

Comment:
I think you need to reassign `test` here so that it is in the scope of inside the loop or else by the time the subtest runs, `test` will have changed.

I think you could also call `t.Parallel()` in the subtest.